src/pages/instances/actions/AttachIsoBtn.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { FC, useState } from "react";
+import { updateInstance } from "api/instances";
+import { LxdInstance } from "types/instance";
+import { useParams } from "react-router-dom";
+import { useNotify } from "@canonical/react-components";
+import { getInstanceEditValues, getInstancePayload } from "util/instanceEdit";
+import { LxdIsoDevice } from "types/device";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import usePortal from "react-useportal";
+import { RemoteImage } from "types/image";
+import SubmitButton from "components/SubmitButton";
+import CustomIsoModal from "pages/images/CustomIsoModal";
+
+interface Props {
+  instance: LxdInstance;
+}
+
+const AttachIsoBtn: FC<Props> = ({ instance }) => {
+  const { project } = useParams<{ project: string }>();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const [isLoading, setLoading] = useState(false);
+
+  const attachedIso = instance.devices["iso-volume"] as
+    | LxdIsoDevice
+    | undefined;
+
+  const ejectIso = () => {
+    setLoading(true);
+    const values = getInstanceEditValues(instance);
+    values.devices = values.devices.filter((device) => {
+      return device.name !== "iso-volume";
+    });
+    const instanceMinusIso = getInstancePayload(
+      instance,
+      values
+    ) as LxdInstance;
+    updateInstance(instanceMinusIso, project ?? "")
+      .then(() =>
+        notify.success(
+          <>
+            ISO <b>{attachedIso?.source ?? ""}</b> ejected
+          </>
+        )
+      )
+      .catch((e: Error) => notify.failure("Detach ISO failed", e))
```

Comment:
Is there a reason why we use "eject" as the verb for the success message but "detach" here? I would use a bit of consistency and change this to:
```suggestion
      .catch((e: Error) => notify.failure("ISO ejection failed", e))
```

Reply:
We had a discussion about it and said it would be consistent to use detach/attach in more places of the UI.

Reply:
Ok, then maybe it would be worth removing all the "eject"s? We have a mixed usage at the moment.

Reply:
I think the latest state of the branch removed them all.